packages/lib/src/interactions/hooks.ts

```diff
@@ -88,17 +109,16 @@ export function useZoomOnWheel(
         // Use `divide` instead of `multiply` by 1 / zoomVector to avoid rounding issues (https://github.com/silx-kit/h5web/issues/1088)
         camera.scale.divide(zoomVector).min(ONE_VECTOR);
       }
-      camera.updateMatrixWorld();
 
-      // Scale the change in position according to the zoom
-      const oldPosition = worldPt.clone();
-      const delta = camera.position.clone().sub(oldPosition);
-      const scaledDelta =
-        sourceEvent.deltaY < 0
-          ? delta.multiply(zoomVector)
-          : delta.divide(zoomVector);
+      // Scale change in position according to zoom
+      const delta = camera.position.clone().sub(worldPt);
+      if (sourceEvent.deltaY < 0) {
+        delta.multiply(zoomVector);
+      } else {
+        delta.divide(zoomVector);
+      }
 
-      moveCameraTo(oldPosition.add(scaledDelta));
+      moveCameraTo(worldPt.clone().add(delta));
```

Comment:
Small refactoring to make the mutations of `delta` and the use of `worldPt` more explicit.